targets/TARGET_STM/trng_api.c

```diff
@@ -63,23 +66,31 @@ void trng_free(trng_t *obj)
     HAL_RNG_DeInit(&obj->handle);
     /* RNG Peripheral clock disable - assume we're the only users of RNG  */
     __HAL_RCC_RNG_CLK_DISABLE();
+
+    users = 0;
```

Comment:
line 36 - uses atomic access, not here when you are setting it?

Reply:
In the free case, I don't have to test the value, I just have to set it to unused, so I preferred to keep it simple: as there is only 1 user, once freed, it will always be 0. 

An alternative is to throw an error in case of unbalanced, (like in sleep manager): 

    core_util_critical_section_enter();
    if (users == 0) {
        core_util_critical_section_exit();
        error("Unbalanced number of trng free (< 0)");
    }
    core_util_atomic_decr_u16(&users, 1);
    core_util_critical_section_exit();

but that seemed overkill to me ...
I can do the change anyway if people think it's of added value.

Reply:
OK, can be as it is.

Why init can be called only once - one owner limitation?

Reply:
yes - one HW only and only one client in MBED in features/mbedtls/platform/src/mbed_trng.c, so I'll stick to it for now.